controllers/secret/secret_controller.go

```diff
@@ -213,6 +223,28 @@ func (r *ReconcileSecret) Reconcile(request reconcile.Request) (reconcile.Result
 	}
 }
 
+// removeDepreciatedAnnotations corrects annotations applied by previous versions of WMCO.
```

Comment:
I suggest renaming this to `removeInvalidAnnotationsFromLinuxNodes` or something to that effect. Please mention the annotation that you are deleting. I don't see any reason to future proof given this is just a helper.